druid/src/widget/padding.rs

```diff
@@ -51,25 +51,37 @@ impl<T> Padding<T> {
     /// use druid::kurbo::Insets;
     ///
     /// let child: Label<()> = Label::new("I need my space!");
-    /// let _: Padding<()> = Padding::new((10.0, 20.0), Label::new("more y than x!"));
+    /// let _: Padding<(), _> = Padding::new((10.0, 20.0), Label::new("more y than x!"));
     /// // equivalent:
-    /// let _: Padding<()> = Padding::new(Insets::uniform_xy(10.0, 20.0), Label::new("ditto :)"));
+    /// let _: Padding<(), _> = Padding::new(Insets::uniform_xy(10.0, 20.0), Label::new("ditto :)"));
     /// ```
     ///
     /// [`kurbo::Insets`]: https://docs.rs/kurbo/0.5.3/kurbo/struct.Insets.html
-    pub fn new(insets: impl Into<Insets>, child: impl Widget<T> + 'static) -> Padding<T> {
+    pub fn new(insets: impl Into<Insets>, child: W) -> Padding<T, W> {
         let insets = insets.into();
         Padding {
             left: insets.x0,
             right: insets.x1,
             top: insets.y0,
             bottom: insets.y1,
-            child: WidgetPod::new(child).boxed(),
+            child: WidgetPod::new(child),
         }
     }
 }
 
-impl<T: Data> Widget<T> for Padding<T> {
+impl<T, W> Padding<T, W> {
+    /// A reference to this widget's child.
+    pub fn child(&self) -> &W {
+        self.child.widget()
+    }
+
+    /// A mutable reference to this widget's child.
+    pub fn child_mut(&mut self) -> &mut W {
+        self.child.widget_mut()
+    }
+}
```

Comment:
I think you can use `widget_wrapper_body` here

Reply:
mm, I'd completely forgotten about that. It looks like it's only implemented for types that don't contain a `WidgetPod`? The intent when that was merged was just to facilitate robert's experiments with bindings (or maybe scope?) and I'm not sure if it makes sense as a general mechanism? (In particular it's hard to discover, and requires having the trait in scope).

Maybe @rjwittams can chime in.

Reply:
> It looks like it's only implemented for types that don't contain a WidgetPod? 

there is also `widget_wrapper_pod_body`.

> I'm not sure if it makes sense as a general mechanism?

I am not sure either but implementing a trait feels better to me.

> In particular it's hard to discover, and requires having the trait in scope

I wish we had ability to mark trait as "important", so that rustdoc highlights it.